### PR TITLE
Add developer documentation for making a release

### PR DESCRIPTION
**Pull request by h-mayorquin**
As in the title.

### CODE REVIEW

**Comment by bendichter** [comment]
I think the new file needs to be an .md or an .rst

**Comment by h-mayorquin** [comment]
And I was thinking that my preview extension was being lazy for not displaying correctly : P

File: docs/developer_guide/making_a_release.rst [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+1. **Format Changelog**:
+
+   - Format and update the changelog.
+   - Ensure correct formatting and add a header to indicate that the changes belong to a specific release and not upcoming ones.
+   - Example: `Commit Example <https://github.com/catalystneuro/neuroconv/commit/2fbea8f05e5bd92c445fcbb6bf24de45330fcbbc>`_
+
+2. **Set the Correct Version for Release**:
+
+   - The development version (the current code on `main`) should be one patch version ahead of the latest PyPI release and therefore ready.
```
Comment (CodyCBakerPhD) [inline docs/developer_guide/making_a_release.rst:14]:
```suggestion
   - The development version (the current code on `main`) should be one patch version ahead of the latest PyPI release and therefore ready for the next step.
```

File: docs/developer_guide/making_a_release.rst [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+   - If a minor version bump is necessary, change it accordingly.
+   - Example: `Version Change Example <https://github.com/catalystneuro/neuroconv/commit/af91f09f300cb36ba4fee483196c8cb492c180ae>`_
+
+3. **Perform Checks**:
+
+   - Ensure that no requirement files include pointers to non-standard branches (e.g., GitHub links).
```
Comment (CodyCBakerPhD) [inline docs/developer_guide/making_a_release.rst:20]:
```suggestion
   - Ensure that no requirement files include pointers to `git`-based dependencies (including specific branches or commit hashes). All dependencies for a PyPI release should point to the released package versions that are available on conda-forge or PyPI.
```

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**